src/Stryker.Core/Stryker.Core/Reporters/Json/JsonReport.cs

```diff
@@ -58,14 +59,20 @@ public static JsonReport Build(StrykerOptions options, IReadOnlyProjectComponent
             return files;
         }
 
-        private IDictionary<string, SourceFile> GenerateFileReportComponents(IReadOnlyFileLeaf fileComponent)
-        {
-            return new Dictionary<string, SourceFile> { { fileComponent.FullPath, new SourceFile(fileComponent) } };
-        }
+        private static IDictionary<string, SourceFile> GenerateFileReportComponents(IReadOnlyFileLeaf fileComponent) => new Dictionary<string, SourceFile> { { fileComponent.RelativePath, new SourceFile(fileComponent) } };
 
-        private void Merge<TTo, TFrom>(IDictionary<TTo, TFrom> to, IDictionary<TTo, TFrom> from)
+        private void AddTestFiles(TestProjectsInfo testProjectsInfo)
         {
-            from.ToList().ForEach(x => to[x.Key] = x.Value);
+            if (testProjectsInfo is not null && testProjectsInfo.TestFiles is not null)
```

Comment:
```suggestion
            if (testProjectsInfo?.TestFiles is not null)
```
Something like this should work I guess.